src/server/repositories/UrlRepository.ts

```diff
@@ -54,28 +58,57 @@ export class UrlRepository implements UrlRepositoryInterface {
   }
 
   public create: (
-    properties: { userId: number; shortUrl: string; longUrl?: string },
+    properties: {
+      userId: number
+      shortUrl: string
+      longUrl?: string
+      tags?: string[]
+    },
     file?: StorableFile,
   ) => Promise<StorableUrl> = async (properties, file) => {
     const newUrl = await sequelize.transaction(async (t) => {
-      await Url.create(
-        {
-          ...properties,
-          longUrl: file
-            ? this.fileBucket.buildFileLongUrl(file.key)
-            : properties.longUrl,
-          isFile: !!file,
-        },
-        {
-          transaction: t,
-        },
-      )
+      const tagStrings = properties.tags
+        ? properties.tags.join(tagSeparator)
+        : ''
+      const urlStaticDTO = {
+        ...properties,
+        longUrl: file
+          ? this.fileBucket.buildFileLongUrl(file.key)
+          : properties.longUrl,
+        isFile: !!file,
+        tagStrings,
+      }
+      const url = await Url.create(urlStaticDTO, {
+        transaction: t,
+      })
+      if (properties.tags) {
+        const tagCreationResponses = await Promise.all(
+          properties.tags.map(async (tag) => {
+            return Tag.findOrCreate({
+              where: {
+                tagString: tag,
+                tagKey: tag.toLowerCase(),
+              },
+              transaction: t,
+            })
+          }),
+        )
```

Comment:
Would it be better to abstract this function into the `TagRepository`? Logically speaking, it might be more consistent and intuitive if the creation of tags happens in the `TagRepository` instead.

It could be a really thin method `tagRepository.findOrCreateTag(tag: string, t: transaction)` that just abstracts over the `Tag.findOrCreate` method, like the one we use in `UserRepository`:

```
public findOrCreateWithEmail: (email: string) => Promise<StorableUser> = (
    email,
  ) => {
    return User.findOrCreate({ where: { email } }).then(([user, _]) => user)
  }
```

Reply:
agreed, will do this